lib/str-repr/tests/test_decode.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2015-2016 Balabit
+ * Copyright (c) 2015-2016 Balázs Scheidler
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#include "str-repr/decode.h"
+#include "testutils.h"
+
```

Comment:
Well, this test predates criterion, and I did convert one complicated testcase in this branch.

Yeah, some of this code is that old. Could this go in with these tests please? Pretty please?

promise to convert the next tests to criterion I am touching.